WordPress/gutenberg
# Site editor sidebar: consolidate rename component and use TemplateActions component for patterns single view

## What's happening here?

This PR:

- unifies the export/duplicate/rename/delete actions dropdown into a single actions component 
- renames 'Template Actions' to 'Pattern actions' in anticipation of the [great unification](https://github.com/WordPress/gutenberg/pull/54271#issuecomment-1760612399).


https://github.com/WordPress/gutenberg/assets/6458278/66221a87-82b1-42fb-924b-fb8c02a31aa6



This PR builds on the work in:

- https://github.com/WordPress/gutenberg/pull/54173#pullrequestreview-1610241523

## TODO

- [ ] Write some e2e tests

## Why? 

#54173 added rename/delete actions to Template parts. Why should patterns miss out?

## How?

- Rejigging `<TemplateActions />` (now renamed to `<PatternActions />`) to handle Pattern post types
- Consolidating template and pattern rename functionality into one `<RenameMenuItem />` component
- Consolidating template and pattern duplicate functionality into one `<DuplicateMenuItem />` component

Note: I haven't looked at the new dataview yet. It might require some refactoring if it exits experimental:
<img width="611" alt="Screenshot 2023-12-04 at 4 22 50 pm" src="https://github.com/WordPress/gutenberg/assets/6458278/836d82ff-e1f6-4455-a1f8-c626ffa91c78">


## Testing

1. Create a custom pattern in the Site Editor
2. From the sidebar, navigate to Patterns > My Patterns > Your new custom pattern. Click on your pattern's title so that the single pattern view appears in the side bar.
3. Ensure that the rename, duplicate, clear, export and delete functionality works
4. Do the same for patterns on the main patterns page, `/wp-admin/site-editor.php?path=%2Fpatterns`
5. Test out custom templates and template parts to ensure there are no regressions
6. Check that single pattern renaming et.al from the command center still works as per https://github.com/WordPress/gutenberg/pull/55188
7. Check that the actions work on "Manage all" pages for template and template parts, e.g., `/wp-admin/site-editor.php?path=%2Fwp_template%2Fall` and `/wp-admin/site-editor.php?path=%2Fwp_template_part%2Fall`
